lib/spack/docs/testing_guide.rst

```diff
@@ -0,0 +1,299 @@
+
+The logs contain one test case per package being built.
+
+^^^^^
+Junit
+^^^^^
+
+To generate logs in `JUnit <http://junit.org/>`_ format, use
+``--log-format=junit``:
+
+.. code-block:: console
+
+   $ spack install --log-format=junit <spec>
+
+The `Jenkins <https://jenkins.io/>`_, `Bamboo
+<https://www.atlassian.com/software/bamboo>`_ and many other tools can
+use JUnit.
+
+^^^^^
+CDash
+^^^^^
+
+`CDash <https://www.cdash.org>`__ is an open source web application from
+`Kitware <https://www.kitware.com>`_ that displays the results of
+software builds and test runs.  Spack can generate output for CDash using
+either of these options to ``spack install``:
+
+  * ``--log-format=cdash-simple``
+  * ``--log-format=cdash-complete``
+
+For example:
+
+.. code-block:: console
+
+   $ spack install --log-file report --log-format=cdash-simple libelf
+
+This will generate a file called ``report.xml`` that contains a
+description of a build of ``libelf``:
+
+.. code-block:: xml
+
+    <?xml version="1.0" ?>
+    <Site BuildName="libelf@0.8.12%gcc@6.2.0 arch darwin-elcapitan-x86_64 /qmbmcez"
+          BuildStamp="20172803-12:00:08-Experimental"
+          CompilerName="gcc"
+          CompilerVersion="6.2.0"
+          Hostname="OS X 10.11.6"
+          Name="atala.llnl.gov"
+          OSName="OS X 10.11.6">
+        <Build>
+            <StartDateTime>Mar 28 12:00 PDT</StartDateTime>
+            <StartBuildTime>1490727615</StartBuildTime>
+            <BuildCommand>spack install</BuildCommand>
+            <Log Encoding="base64">
+                <!-- ... long build log text goes here ... -->
+            </Log>
+            <EndDateTime>Mar 28 12:00 PDT</EndDateTime>
+            <EndBuildTime>1490727615</EndBuildTime>
+            <ElapsedMinutes>0</ElapsedMinutes>
+        </Build>
+    </Site>
+
+Spack uses a short version of each package's spec as the CDash build
+name, which will be shown in the dashboard.
+
+For more detailed output, you can use the ``cdash-complete`` format:
+
+.. code-block:: console
+
+   $ spack install --log-file report --log-format=cdash-complete libelf
+
+This will create separate files: ``report.build.xml``,
+``report.configure.xml``, and ``report.test.xml``, for the build,
+configure, and tests steps, respectively.
+
+If you want to upload these fils to a CDash instance, you can use ``curl``:
+
+.. code-block:: console
+
+   $ curl --upload-file report.build.xml https://example.com/cdash/submit.php?project=<projectname>
+   $ curl --upload-file report.configure.xml https://example.com/cdash/submit.php?project=<projectname>
+   $ curl --upload-file report.test.xml https://example.com/cdash/submit.php?project=<projectname>
+
+Spack can also automate this step for you as part of the ``spack
+test-suite`` command described in the next section.
+
+
+.. _cmd-spack-test-suite:
+
+---------------------
+CDash test suites
+---------------------
+
+The ``spack test-suite`` command reads in a specialy formatted YAML file
```

Comment:
Are you going to keep `test-suite`?  I think it needs a rework, but it would make sense if it is part of what drives the CI.  The issue I see is that it's sequential, and if we want to test 2700 packages quickly, we need to farm them out to separate runners.

Reply:
Well, I was going to keep it, thinking it was in some ways separate from the release process.  I'll give it another look, but like you say, we'll need to parallelize the builds.  My feeling is that we'll want to "fan out" to as many little jobs as possible so we have maximum flexibility in how we distribute them.  I have spent quite a bit of time discussing approaches in this area with @opadron, so it will be good when we can meet and discuss them.  I'll make another comment on our current thinking along those lines somewhere else.

Reply:
I think the `test-suite` command can be retired and we can start from scratch, given that you guys have much more flexible ideas about this.